lib/engine/game/g_1849/step/buy_train.rb

```diff
@@ -34,6 +37,34 @@ def can_sell?(entity, bundle)
 
             super
           end
+
+          def buyable_trains(entity)
+            # Cannot buy E-train without E-token
+            trains_to_buy = super
+
+            trains_to_buy = trains_to_buy.reject { |t| t.name == 'E' }
+            trains_to_buy << e_train if @game.electric_dreams? && can_buy_e?(entity)
```

Comment:
It doesn't make sense to reject and then add back in the e train. instead do something like
```
if !@game.electric_dreams? || !can_buy_e?(entity)
  trains_to_buy.reject! { |t| t.name == "E" }
```
trains_to_buy